server/ec2spotmanager/common/prices.py

```diff
@@ -12,66 +12,21 @@
 
 @contact:    choller@mozilla.com
 '''
-import datetime
 
-import botocore
-import boto3
-from django.utils import timezone
 
-
-# Blacklist zones that currently don't allow subnets to be set on them
-# until we found a better way to deal with this situation.
-zone_blacklist = ["us-east-1a", "us-east-1f"]
-
-
-# This function must be defined at the module level so it can be pickled
-# by the multiprocessing module when calling this asynchronously.
-def get_spot_price_per_region(region_name, aws_key_id, aws_secret_key, instance_types=None):
-    '''Gets spot prices of the specified region and instance type'''
-    prices = {}  # {instance-type: region: {az: [prices]}}}
-
-    now = timezone.now()
-
-    # TODO: Make configurable
-    spot_history_args = {
-        'Filters': [{'Name': 'product-description', 'Values': ['Linux/UNIX']}],
-        'StartTime': now - datetime.timedelta(hours=6)
-    }
-    if instance_types is not None:
-        spot_history_args['InstanceTypes'] = instance_types
-
-    cli = boto3.client('ec2', region_name=region_name, aws_access_key_id=aws_key_id,
-                       aws_secret_access_key=aws_secret_key)
-    paginator = cli.get_paginator('describe_spot_price_history')
-    try:
-        for result in paginator.paginate(**spot_history_args):
-            for price in result['SpotPriceHistory']:
-                if price['AvailabilityZone'] in zone_blacklist:
-                    continue
-                (prices
-                 .setdefault(price['InstanceType'], {})
-                 .setdefault(region_name, {})
-                 .setdefault(price['AvailabilityZone'], [])
-                 .append(float(price['SpotPrice'])))
-    except botocore.exceptions.EndpointConnectionError as exc:
-        raise RuntimeError("Boto connection error: %s" % (exc,))
-
-    return prices
-
-
-def get_spot_prices(regions, aws_key_id, aws_secret_key, instance_types=None, use_multiprocess=False):
+def get_spot_prices(self, regions, instance_types=None, use_multiprocess=False):
```

Comment:
This is a function, not a method. You can't use `self` here. Does this file need to exist at all anymore?

Reply:
Fixed. This method doesn't seem like it belongs in the interface.